### PR TITLE
feat: new User Page

### PR DESCRIPTION
**Pull request by DrShpongle**
This draft PR represents the new `user` page (under `/user-new` route for now). It contains all parts of current User page, but withing categories tabs. Creating/developing new/existing widgets can go with farther PR's.

References:
1. https://excalidraw.com/#room=c2ce7e96050d89fce7af,DMO-WuDkf17wldbEPTtNLA
2. https://github.com/eggheadio/egghead-next/discussions/1212

<details>
  <summary>Desktop:</summary>
  
</details>
<details>
  <summary>Mobile:</summary>
  
</details>

vs tabs GIF

### CODE REVIEW

File: src/components/login-required.tsx [hunk 1]
```diff
@@ -10,15 +10,17 @@ const LoginRequired: FunctionComponent<LoginRequiredParams> = ({
...
   loginRequired = loginRequired || !token
   const router = useRouter()
   React.useEffect(() => {
+    setMounted(true)
     if (loginRequired) {
       router.push('/login')
     }
   }, [loginRequired])
-  return <>{!loginRequired && children}</>
+  return <>{!loginRequired && mounted && children}</>
```
Comment (DrShpongle) [inline src/components/login-required.tsx:23]: fixes hydration warning

File: src/components/pages/user/components/avatar-form.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        <div className="w-32 h-32 rounded-full flex overflow-hidden shrink-0 border border-gray-200">
+          <ImageWithPlaceholder
+            src={`https:${avatarUrl}`}
+            width={256}
+            height={256}
+            loading="eager"
+          />
+        </div>
+        {/* Need server functionality implemented */}
+        {/* <label
```
Comment (DrShpongle) [inline src/components/pages/user/components/avatar-form.tsx:17]: commented out code will be needed after functionality implemented

File: src/pages/user-new/index.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  //   label: 'Membership',
+  //   component: () => {
+  //     return (
+  //       <div>
+  //         <div>Membership</div>
+  //       </div>
+  //     )
+  //   },
+  // },
+  {label: 'Bookmarks', component: <BookmarksTabContent />},
```
Comment (DrShpongle) [inline src/pages/user-new/index.tsx:135]: decided to set up separate tab for Bookmarks, it will be more useful imo

**Comment by zacjones93** [comment]
This looks awesome, I want to get in there with @jbranchaud and hook up the missing functionality: update avatar, preferred name, disconnect from github

Dark mode styles need a couple tweaks:

image

What do you think of no border on the elements? a little less visual noise when the structure is already very clear

Build the portfolio you need to be a badass web developer   egghe

**Comment by DrShpongle** [comment]
> 
> What do you think of no border on the elements? a little less visual noise when the structure is already very clear
> 

Image 2022-10-03 at 11 16 10

It looks less "outlined" for me.

**Comment by DrShpongle** [comment]
Adjusted dark mode styles:

<details>
  <summary>Light:</summary>
  
</details>
<details>
  <summary>Dark:</summary>
  
</details>

**Pull request merged**